src/home_robot/home_robot/navigation_planner/discrete_planner.py

```diff
@@ -138,49 +145,110 @@ def plan(
             int(start_x * 100.0 / self.map_resolution - gy1),
         ]
         start = pu.threshold_poses(start, obstacle_map.shape)
+        start = np.array(start)
+
+        if debug:
+            print("Goals found:", np.any(goal_map > 0))
 
         self.curr_pose = [start_x, start_y, start_o]
         self.visited_map[gx1:gx2, gy1:gy2][
             start[0] - 0 : start[0] + 1, start[1] - 0 : start[1] + 1
         ] = 1
 
+        # Check collisions if we have just moved and are uncertain
         if self.last_action == DiscreteNavigationAction.MOVE_FORWARD:
             self._check_collision()
 
         # High-level goal -> short-term goal
         # t0 = time.time()
+        # Extracts a local waypoint
+        # Defined by the step size - should be relatively close to the robot
         short_term_goal, closest_goal_map, replan, stop = self._get_short_term_goal(
             obstacle_map, np.copy(goal_map), start, planning_window
         )
+        # Short term goal is in cm, start_x and start_y are in m
+        if debug:
+            print("Current pose:", start)
+            print("Short term goal:", short_term_goal)
+            dist_to_short_term_goal = np.linalg.norm(
+                start - np.array(short_term_goal[:2])
+            )
+            print("Distance:", dist_to_short_term_goal)
+            print("Replan:", replan)
+            print()
         # t1 = time.time()
         # print(f"[Planning] get_short_term_goal() time: {t1 - t0}")
 
         # We were not able to find a path to the high-level goal
         if replan:
-            # Clean collision map
-            self.collision_map *= 0
-
-            # Reduce obstacle dilation
-            if self.curr_obs_dilation_selem_radius > 1:
-                self.curr_obs_dilation_selem_radius -= 1
-                self.obs_dilation_selem = skimage.morphology.disk(
-                    self.curr_obs_dilation_selem_radius
-                )
+            print("Could not find a path to the high-level goal. Stopping.")
+            action = DiscreteNavigationAction.STOP
+            # TODO Calling the STOP action here will cause the agent to try grasping
+            #   we need different STOP_SUCCESS and STOP_FAILURE actions
+            breakpoint()
```

Comment:
remove the breakpoint? and the dead code underneath?

Reply:
changed a bit but this is WIP

Reply:
should be ok now, will just terminate